frontend/discourse/app/components/header.gjs

```diff
@@ -184,10 +183,24 @@ export default class GlimmerHeader extends Component {
   }
 
   @action
-  toggleUserMenu(value) {
-    this.header.userVisible = value ?? !this.header.userVisible;
+  async toggleUserMenu(value) {
+    const wasVisible = this.header.userVisible;
+    const willBeVisible = value ?? !wasVisible;
+    const isClosing = wasVisible && !willBeVisible;
+
+    // If closing and on desktop, animate before closing
+    if (isClosing && this.site.desktopView) {
+      const menuPanel = document.querySelector(".menu-panel.drop-down");
```

Comment:
This could be brittle, could you pass element reference like in the hamburger one:
https://github.com/discourse/discourse/blob/f40873a4155f4cdde59f173fe861faf4c5ed42f6/frontend/discourse/app/components/header/hamburger-dropdown-wrapper.gjs#L28